From f82b3ea3b8ba3bf5098f1ff9acf1408aac5e2e28 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 17 Aug 2020 10:31:41 +0800
Subject: [PATCH] driver: net: lan78xx: fix building issue introduced by
 merging code from v5.4/standard/base

There are some raspberry pi sdk patches applied to the driver
lan78xx.c. So, when merge code from v5.4/standard/base, there are
conflicts, and not resolve them completely.
Therefore, adjust code manually so that fix building issue.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/usb/lan78xx.c | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index 04181af2409b..f2a1067ed9a7 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -3773,23 +3773,8 @@ static int lan78xx_probe(struct usb_interface *intf,
 	netdev->max_mtu = MAX_SINGLE_PACKET_SIZE;
 	netif_set_gso_max_size(netdev, MAX_SINGLE_PACKET_SIZE - MAX_HEADER);
 
-	if (intf->cur_altsetting->desc.bNumEndpoints < 3) {
-		ret = -ENODEV;
-		goto out3;
-	}
-
-	dev->ep_blkin = (intf->cur_altsetting)->endpoint + 0;
-	dev->ep_blkout = (intf->cur_altsetting)->endpoint + 1;
-	dev->ep_intr = (intf->cur_altsetting)->endpoint + 2;
-
-	dev->pipe_in = usb_rcvbulkpipe(udev, BULK_IN_PIPE);
-	dev->pipe_out = usb_sndbulkpipe(udev, BULK_OUT_PIPE);
-
-	dev->pipe_intr = usb_rcvintpipe(dev->udev,
-					dev->ep_intr->desc.bEndpointAddress &
-					USB_ENDPOINT_NUMBER_MASK);
 	if (int_urb_interval_ms <= 0)
-		period = dev->ep_intr->desc.bInterval;
+		period = ep_intr->desc.bInterval;
 	else
 		period = int_urb_interval_ms * INT_URB_MICROFRAMES_PER_MS;
 
-- 
2.17.1

